komodo/show_version.py

```diff
@@ -0,0 +1,133 @@
+    """
+    Read an INI file (aka config file) that does not have sections.
+    Creates a temporary 'MAIN' section and pass bask its contents.
+    """
+    config = configparser.ConfigParser()
+    with open(fname, "rt") as f:
+        config.read_string("[MAIN]\n" + f.read())
+    return dict(config["MAIN"])
+
+
+def guess_komodoenv_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path, but from a
+    komodoenv environment. These environments overwrite the environment
+    variable $KOMODO_RELEASE and $PATH, but they store the original path
+    in the komodoenv at 'root/pyvenv.cfg'.
+
+    Args:
+        release (str): The name of the release, e.g. from $KOMODO_RELEASE.
+
+    Returns:
+        str: The path to the release, where the 'root/bin' folder is.
+    """
+    config = read_config(Path(release) / "root" / "pyvenv.cfg")
+    path, *_ = config["home"].split("/root/bin")
+    return Path(path)
+
+
+def guess_komodo_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path in an ordinary
+    komodo environment. E.g., the release may be something like
+    '2023.01.02-py38' but the 'real' release (where the release manifest
+    is stored) might be platform-specific, e.g. '2023.01.02-py38-rhel7'.
+    The real path is in the PATH, so we try to get it from there.
+
+    Args:
+        release (str): The name of the release, e.g. from $KOMODO_RELEASE.
+
+    Returns:
+        str: The path to the release, where the 'root/bin' folder is.
+    """
+    paths = os.environ["PATH"]
+    pattern = re.compile(rf"(.*?{release}.*?)/root/bin")
+    result = pattern.search(paths)
+    if result is not None:
+        (path,) = result.groups()
+    else:
+        raise Exception("Could not sense the path to the release.")
+    return Path(path)
+
+
+def get_version(pkg: str, fpath: Optional[str] = None) -> str:
+    """
+    Get the release number (or git commit hash) for a package in a
+    komodo release file. If no file is specified, the current release
+    is used. If no environment is active, the path to the release
+    must be specified.
+
+    Args:
+        pkg (str): The name of the package to get the version for.
+        fpath (str): Optional. The path to a release 'manifest' file.
+
+    Returns:
+        str: The version number or git hash of the version.
+    """
+    if fpath is not None:
+        path = Path(fpath)
+        release_file = path.stem
+        path = path.parent
+    else:
+        release = guess_release()
+        if (Path(release) / "komodoenv.conf").is_file():
+            path = guess_komodoenv_path(release)
+        else:
+            path = guess_komodo_path(release)
+        release_file = path.parts[-1]
+
+    with open(path / release_file, "rt") as f:
+        packages = yaml.safe_load(f)
+    package = packages.get(pkg)
+
+    if package is None:
+        raise ValueError(f"The package {pkg} is not in the release.")
```

Comment:
Add the content of the `release` string to the error message?

Reply:
Good one.